micrometer-core/src/test/java/io/micrometer/core/instrument/step/StepMeterRegistryTest.java

```diff
@@ -342,6 +321,57 @@ void publishOnCloseCrossesStepBoundary() {
         assertThat(registry.publishedFunctionTimerTotals.pop()).isEqualTo(53);
     }
 
+    @Test
+    @Issue("#3863")
+    void shouldPublishLastCompletedStepOnClosingRollover() {
```

Comment:
Nit pick on the test name: how about `shouldPublishLastCompletedStepWhenClosingBeforeScheduledPublish`?